src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
Because you still need to call it in this block:

```
121     // Start piping from client to proxy                                                                         | 15     Label string
122     go func() {                                                                                                  | 16
123         if _, err := io.Copy(connOut, clientConn); err != nil {                                                  | 17     // Weight: determines how often this Dialer is used relative to the other
124             log.Debugf("Error piping data from client to proxy: %s", err)                                        | 18     // Dialers on the balancer.
125         }                                                                                                        | 19     Weight int
126         closeOnce.Do(closeConns)                                                                                 | 20
127     }()
```

It actually should pass the full closeOnce.Do(closeConns) as a func though, so something like:

```
pipeData(clientConn, connOut, req, func() {closeOnce.Do(closeConns)})
```

Does that make sense or am I missing something? I think you still need to call that from the go routine.